src/spikeinterface/postprocessing/correlograms.py

```diff
@@ -595,3 +604,418 @@ def _compute_correlograms_one_segment_numba(
+    ):
+        params = dict(
+            window_ms=window_ms,
+            bin_ms=bin_ms,
+            num_firing_rate_quantiles=num_firing_rate_quantiles,
+            smoothing_factor=smoothing_factor,
+        )
+
+        return params
+
+    def _select_extension_data(self, unit_ids):
+        # filter metrics dataframe
+        unit_indices = self.sorting_analyzer.sorting.ids_to_indices(unit_ids)
+        new_acgs_3d = self.data["acgs_3d"][unit_indices]
+        new_firing_quantiles = self.data["firing_quantiles"][unit_indices]
+        new_bins = self.data["bins"][:]
+        new_data = dict(acgs_3d=new_acgs_3d, firing_quantiles=new_firing_quantiles, bins=new_bins)
+        return new_data
+
+    def _merge_extension_data(
+        self, merge_unit_groups, new_unit_ids, new_sorting_analyzer, censor_ms=None, verbose=False, **job_kwargs
+    ):
+        new_sorting = new_sorting_analyzer.sorting
+
+        acgs_3d, firing_rate_quantiles, _ = _compute_acgs_3d(
+            new_sorting,
+            unit_ids=new_unit_ids,
+            window_ms=self.params["window_ms"],
+            bin_ms=self.params["bin_ms"],
+            num_firing_rate_quantiles=self.params["num_firing_rate_quantiles"],
+            smoothing_factor=self.params["smoothing_factor"],
+            **job_kwargs,
+        )
+
+        new_unit_ids_indices = new_sorting.ids_to_indices(new_unit_ids)
+        old_unit_ids = [unit_id for unit_id in new_sorting_analyzer.unit_ids if unit_id not in new_unit_ids]
+        old_unit_ids_indices = new_sorting.ids_to_indices(old_unit_ids)
+
+        new_acgs_3d = np.zeros((len(new_sorting.unit_ids), acgs_3d.shape[1], acgs_3d.shape[2]))
+        new_firing_quantiles = np.zeros((len(new_sorting.unit_ids), firing_rate_quantiles.shape[1]))
+
+        new_acgs_3d[new_unit_ids_indices, :, :] = acgs_3d
+        new_acgs_3d[old_unit_ids_indices, :, :] = self.data["acgs_3d"][old_unit_ids_indices, :, :]
+
+        new_firing_quantiles[new_unit_ids_indices, :] = firing_rate_quantiles
+        new_firing_quantiles[old_unit_ids_indices, :] = self.data["firing_quantiles"][old_unit_ids_indices, :]
+
+        new_data = dict(
+            acgs_3d=new_acgs_3d,
+            firing_quantiles=new_firing_quantiles,
+            bins=self.data["bins"],
+        )
+        return new_data
+
+    def _run(self, verbose=False, **job_kwargs):
+        acgs_3d, firing_quantiles, bins = _compute_acgs_3d(self.sorting_analyzer.sorting, **self.params, **job_kwargs)
+        self.data["firing_quantiles"] = firing_quantiles
+        self.data["acgs_3d"] = acgs_3d
+        self.data["bins"] = bins
+
+    def _get_data(self):
+        return self.data["acgs_3d"], self.data["firing_quantiles"], self.data["bins"]
+
+
+def _compute_acgs_3d(
+    sorting: BaseSorting,
+    unit_ids=None,
+    window_ms: float = 50.0,
+    bin_ms: float = 1.0,
+    num_firing_rate_quantiles: int = 10,
+    smoothing_factor: int = 250,
+    **job_kwargs,
+):
+    """
```

Comment:
If this is internal I would not put long doc here.
Or maybe I would rename this compute_acgs_3d_sorting and expose it to the API.